astroid/nodes/_base_nodes.py

```diff
@@ -213,7 +213,7 @@ class MultiLineWithElseBlockNode(MultiLineBlockNode):
     def blockstart_tolineno(self):
         return self.lineno
 
-    def _elsed_block_range(self, lineno, orelse, last=None):
+    def _elsed_block_range(self, lineno, orelse, last=None) -> tuple[int, int]:
```

Comment:
```suggestion
    def _elsed_block_range(self, lineno, orelse, last=None) -> tuple[int, int | None]:
```

Noticed a few times were you omitted the optional `None`. Please take a look at the changes again. Will comment on one or two, but quite likely I'll miss some.